tests/load/test_read_interfaces.py

```diff
@@ -472,6 +436,138 @@ def test_schema_arg(populated_pipeline: Pipeline) -> None:
     assert "items" in dataset.schema.tables
 
 
+@pytest.mark.no_load
+@pytest.mark.essential
+@pytest.mark.parametrize(
+    "populated_pipeline",
+    configs,
+    indirect=True,
+    ids=lambda x: x.name,
+)
+def test_ibis_expression_relation(populated_pipeline: Pipeline) -> None:
+    # NOTE: we could generalize this with a context for certain deps
+
+    # now we should get the more powerful ibis relation
+    dataset = populated_pipeline._dataset()
+    total_records = _total_records(populated_pipeline)
+
+    items_table = dataset["items"]
+    double_items_table = dataset["double_items"]
+
+    # check full table access
+    df = items_table.df()
+    assert len(df.index) == total_records
+
+    df = double_items_table.df()
+    assert len(df.index) == total_records
+
+    # check limit
+    df = items_table.limit(5).df()
+    assert len(df.index) == 5
+
+    # check chained expression with join, column selection, order by and limit
+    joined_table = (
```

Comment:
you do not need to test all the expressions but you should test various expression forms:
1. those that return Table (like here)
2. those that select a column
3. a single column getter with . notation
4. a free form filters:
```py
expr = penguins.bill_length_mm > 37.0
```
5. materializations like their dataframe and arrow getters
6. there are also expressions that add and remove columns to schema
7. expressions that return Expr (not Table - those do some weird things, I'm not sure SQL can be generated for them)?

Reply:
I have added quite a few tests from the "ibis for sql users" page in the ibis docs. changed a few things in the ibisrelation implementation to make this work. I think most cases should actually work now. Let me know if you think something is missing there or you have an idea what to test next.

Reply:
I ammended it a bit more, I think it should be quite good now.